### ISSUE TITLE
"Table ... has no Fields associated with it" for all tables

### ISSUE THREAD

**Issue by skyler** [metabase/metabase#9428]
- *Repeatable steps to reproduce the issue*

I click **Ask a question** -> **Custom** -> select my data source (BigQuery) -> select my table -> click **Get Answer**, I receive this error:

> **There was a problem with your question**
> 
> Most of the time this is caused by an invalid selection or bad input value. Double check your inputs and retry your query.

I click **Show error details** and see this:

> **Here's the full error message**
> `Table 'my_cool_table' has no Fields associated with it.`

This occurs for the majority of my tables.

I'm running 4 replicas of Metabase in Kubernetes (Google GKE). This morning, all 4 `Pod`s showed output similar to the below. This lead me to believe that all table information was sync'd:

```
#################################################################
# Completed analyze on my_cool_company-dw
# Start: 2019-02-15T11:00:09.214Z
# End: 2019-02-15T11:00:14.731Z
# Duration: 6 s
# ---------------------------------------------------------------
# Completed step 'fingerprint-fields'
# Start: 2019-02-15T11:00:09.214Z
# End: 2019-02-15T11:00:13.598Z
# Duration: 4 s
# Fingerprint updates attempted 59, updated 0, no data found 59, failed 0
# ---------------------------------------------------------------
# Completed step 'classify-fields'
# Start: 2019-02-15T11:00:13.598Z
# End: 2019-02-15T11:00:14.173Z
# Duration: 575 ms
# Total number of fields classified 0, 0 failed
# ---------------------------------------------------------------
# Completed step 'classify-tables'
# Start: 2019-02-15T11:00:14.174Z
# End: 2019-02-15T11:00:14.731Z
# Duration: 557 ms
# Total number of tables classified 284, 0 updated
#################################################################

02-15 11:00:14 INFO sync.util :: FINISHED: Analyze data for bigquery Database 2 'my_cool_company-dw' (6 s)
```

After selecting a table, I see exceptions like this via a Metabase `Pod`'s stdout:

```java
02-15 17:05:17 WARN metabase.query-processor :: {:status :failed,
 :class java.lang.Exception,
 :error "Table 'my_cool_company__my_cool_table' has no Fields associated with it.",
 :stacktrace
 ("--> query_processor.middleware.add_implicit_clauses$fn__30627$add_implicit_fields__30632$fn__30638.invoke(add_implicit_clauses.clj:81)"
  "query_processor.middleware.add_implicit_clauses$fn__30627$add_implicit_fields__30632.invoke(add_implicit_clauses.clj:64)"
  "query_processor.middleware.add_implicit_clauses$fn__30721$add_implicit_mbql_clauses__30726$fn__30732.invoke(add_implicit_clauses.clj:107)"
  "query_processor.middleware.add_implicit_clauses$fn__30721$add_implicit_mbql_clauses__30726.invoke(add_implicit_clauses.clj:105)"
  "query_processor.middleware.add_implicit_clauses$maybe_add_implicit_clauses.invokeStatic(add_implicit_clauses.clj:116)"
  "query_processor.middleware.add_implicit_clauses$maybe_add_implicit_clauses.invoke(add_implicit_clauses.clj:114)"
  "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__30830.invoke(add_row_count_and_status.clj:15)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__32181.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__35350.invoke(resolve_driver.clj:15)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31376$fn__31377.invoke(bind_effective_timezone.clj:9)"
  "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
  "util.date$call_with_effective_timezone.invoke(date.clj:77)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31376.invoke(bind_effective_timezone.clj:8)"
  "query_processor.middleware.store$initialize_store$fn__37567$fn__37568.invoke(store.clj:11)"
  "query_processor.store$do_with_new_store.invokeStatic(store.clj:34)"
  "query_processor.store$do_with_new_store.invoke(store.clj:30)"
  "query_processor.middleware.store$initialize_store$fn__37567.invoke(store.clj:10)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__31798.invoke(cache.clj:127)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__31846.invoke(catch_exceptions.clj:64)"
  "query_processor$process_query.invokeStatic(query_processor.clj:213)"
  "query_processor$process_query.invoke(query_processor.clj:209)"
  "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:322)"
  "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:316)"
  "query_processor$fn__37732$process_query_and_save_execution_BANG___37737$fn__37738.invoke(query_processor.clj:354)"
  "query_processor$fn__37732$process_query_and_save_execution_BANG___37737.invoke(query_processor.clj:340)"
  "query_processor$fn__37756$process_query_and_save_with_max_BANG___37761$fn__37762.invoke(query_processor.clj:373)"
  "query_processor$fn__37756$process_query_and_save_with_max_BANG___37761.invoke(query_processor.clj:369)"
  "api.dataset$fn__44259$fn__44262.invoke(dataset.clj:45)"
  "api.common$fn__23158$invoke_thunk_with_keepalive__23163$fn__23164$fn__23165.invoke(common.clj:435)"),
 :query
 {:type "query",
  :query {:source-table 73},
  :parameters [],
  :constraints {:max-results 10000, :max-results-bare-rows 2000},
  :info
  {:executed-by 1,
   :context :ad-hoc,
   :card-id nil,
   :nested? false,
   :query-hash [47, -75, -18, -31, -113, -113, 110, -64, -17, -101, 82, 111, -114, 3, -24, 70, -7, 29, 40, -119, 28, 40, -46, -40, 46, 51, -84, -106, -57, 42, -43, 102],
   :query-type "MBQL"}},
 :preprocessed nil,
 :native nil}

02-15 17:05:17 WARN metabase.query-processor :: Query failure: Table 'my_cool_company__my_cool_table' has no Fields associated with it.
("clojure.core$ex_info.invokeStatic(core.clj:4739)"
 "clojure.core$ex_info.invoke(core.clj:4739)"
 "--> query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:285)"
 "query_processor$assert_query_status_successful.invoke(query_processor.clj:277)"
 "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:323)"
 "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:316)"
 "query_processor$fn__37732$process_query_and_save_execution_BANG___37737$fn__37738.invoke(query_processor.clj:354)"
 "query_processor$fn__37732$process_query_and_save_execution_BANG___37737.invoke(query_processor.clj:340)"
 "query_processor$fn__37756$process_query_and_save_with_max_BANG___37761$fn__37762.invoke(query_processor.clj:373)"
 "query_processor$fn__37756$process_query_and_save_with_max_BANG___37761.invoke(query_processor.clj:369)"
 "api.dataset$fn__44259$fn__44262.invoke(dataset.clj:45)"
 "api.common$fn__23158$invoke_thunk_with_keepalive__23163$fn__23164$fn__23165.invoke(common.clj:435)")
```

- Your browser and the version: Version 71.0.3578.98 (Official Build) (64-bit)
- Your operating system: macOS Mojave 10.14.2 (18C54)
- Your databases: Google BigQuery
- Metabase version: v0.31.2
- Metabase hosting environment: Kubernetes on Google GKE
- Metabase internal database: MySQL 5.7 (Google Cloud SQL)

**Comment by v1shwa** [comment]
+1 Receiving the same error with few of the MongoDB tables.

**Comment by AnonyV** [comment]
Maybe this is a common bug of syncing database schema with metabase.
Hope can help for fixing the bug.
https://github.com/metabase/metabase/issues/7630#issuecomment-466256461

**Comment by beltz02** [comment]
Hi I'm experiencing the same error on Redshift
Metabase version: v0.32.4 
*Running local .jar file on my local win10 machine

**Comment by flamber** [comment]
@beltz02 If you're using Views, then you're likely seeing #6505

**Comment by KD76** [comment]
+1 I'm experiencing the same issues.

First I thought it was due to nested objects. Then, it to be because of `_id` presence in nested objects. Finally, it seems like nested objects with same attribute key than parent let metabase confused. **I don't know if it's the (only) cause of this issue.**

There's a way to counterfeit this issue :

1. Go to Administration > Data Model
2. Select your targeted database
3. Select the table in which you have the issue
4. Switch visibility to "hidden", then set it back.
5. Reload your tab
6. The table columns are now loaded

I don't know if you have to do this for every schema change in your mongo database.

**Comment by javiSwift** [comment]
I'm experiencing the same issues, and all tables that are getting metabase confused contain the field `_id` in a nested object, so it seems to be the problem at least for me.

After following your steps, the fields of the table are now available, so I can query this table without any problem as before.

**Comment by bacus99** [comment]
Same here for latest release.  Old imported table are working, new one no...  Can't figure

{:database_id 2,
 :started_at #t "2020-11-06T13:56:08.835231-05:00[America/Montreal]",
 :error_type :invalid-query,
 :json_query {:database 2, :query {:source-table 28961}, :type "query", :parameters [], :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> query_processor.middleware.add_implicit_clauses$fn__40552$sorted_implicit_fields_for_table__40557$fn__40558.invoke(add_implicit_clauses.clj:42)"
  "query_processor.middleware.add_implicit_clauses$fn__40552$sorted_implicit_fields_for_table__40557.invoke(add_implicit_clauses.clj:36)"
  "query_processor.middleware.add_implicit_clauses$fn__40648$add_implicit_fields__40653$fn__40657.invoke(add_implicit_clauses.clj:87)"
  "query_processor.middleware.add_implicit_clauses$fn__40648$add_implicit_fields__40653.invoke(add_implicit_clauses.clj:80)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_mbql_clauses.invokeStatic(add_implicit_clauses.clj:124)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_mbql_clauses.invoke(add_implicit_clauses.clj:120)"
  "query_processor.middleware.add_implicit_clauses$maybe_add_implicit_clauses.invokeStatic(add_implicit_clauses.clj:134)"
  "query_processor.middleware.add_implicit_clauses$maybe_add_implicit_clauses.invoke(add_implicit_clauses.clj:131)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__40738.invoke(add_implicit_clauses.clj:141)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__41136.invoke(add_source_metadata.clj:105)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__48304.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__45447.invoke(auto_bucket_datetimes.clj:125)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__46423.invoke(resolve_source_table.clj:46)"
  "query_processor.middleware.parameters$substitute_parameters$fn__48089.invoke(parameters.clj:114)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46475.invoke(resolve_referenced.clj:80)"
  "query_processor.middleware.expand_macros$expand_macros$fn__47094.invoke(expand_macros.clj:158)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__41167.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__49025.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48315$fn__48319.invoke(resolve_database_and_driver.clj:33)"
  "driver$do_with_driver.invokeStatic(driver.clj:61)"
  "driver$do_with_driver.invoke(driver.clj:57)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48315.invoke(resolve_database_and_driver.clj:27)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47312.invoke(fetch_source_query.clj:267)"
  "query_processor.middleware.store$initialize_store$fn__49034$fn__49035.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__49034.invoke(store.clj:10)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__46352.invoke(cache.clj:214)"
  "query_processor.middleware.validate$validate_query$fn__49043.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__47437.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__41005.invoke(add_rows_truncated.clj:36)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49010.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__46715.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__48178.invoke(process_userland_query.clj:136)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__46658.invoke(catch_exceptions.clj:174)"
  "query_processor.reducible$async_qp$qp_STAR___39794$thunk__39795.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___39794.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___39803$fn__39806.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___39803.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:237)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:233)"
  "query_processor$fn__49201$process_query_and_save_execution_BANG___49210$fn__49213.invoke(query_processor.clj:249)"
  "query_processor$fn__49201$process_query_and_save_execution_BANG___49210.invoke(query_processor.clj:241)"
  "query_processor$fn__49245$process_query_and_save_with_max_results_constraints_BANG___49254$fn__49257.invoke(query_processor.clj:261)"
  "query_processor$fn__49245$process_query_and_save_with_max_results_constraints_BANG___49254.invoke(query_processor.clj:254)"
  "api.dataset$fn__52570$fn__52573.invoke(dataset.clj:59)"
  "query_processor.streaming$streaming_response_STAR_$fn__37098$fn__37099.invoke(streaming.clj:73)"
  "query_processor.streaming$streaming_response_STAR_$fn__37098.invoke(streaming.clj:72)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)"
  "async.streaming_response$do_f_async$fn__24643.invoke(streaming_response.clj:85)"],
 :context :ad-hoc,
 :**error "No fields found for table \"nexposet\".",**
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data {:table-id 28961, :type :invalid-query},
 :data {:rows [], :cols []}}

My server

{
  "browser-info": {
    "language": "en-CA",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.8+10-LTS",
    "java.vendor": "N/A",
    "java.vendor.url": "https://openjdk.java.net/",
    "java.version": "11.0.8",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.8+10-LTS",
    "os.name": "Linux",
    "os.version": "3.10.0-1127.19.1.el7.x86_64",
    "user.language": "en",
    "user.timezone": "America/Montreal"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "sqlserver",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "8.0.19"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.6.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-10-26",
      "tag": "v0.37.0.2",
      "branch": "release-x.37.x",
      "hash": "ba7be09"
    },
    "settings": {
      "report-timezone": null
    }
  }
}

**Comment by flamber** [comment]
@bacus99 This issue is specific to BigQuery.
Since I can see it's table id `28961`, then I'm guessing you have a very big database. It will take time to do the sync.
You can follow the process in Admin > Troubleshooting > Logs, and manually force a sync via Admin > Databases > (your-db).
I have a feeling that the sync process is simply not done yet - #11354 - and you might also be seeing #13650

**Comment by bacus99** [comment]
Thank you @flamber, yes, it is a 13 GB Postgres DB...  Security and stuff + a 5 GB GLPI DB combined...

I'll try to remove the most I can from sync

Thank you, will get back to you

**Comment by jeff303** [comment]
@skyler , any chance you can share the DDL for one of the BigQuery tables in question?

**Comment by skyler** [comment]
@jeff303 no, sorry, it's been two years or so since I filed this issue and I don't have that info available anymore.

**Comment by jeff303** [comment]
> @jeff303 no, sorry, it's been two years or so since I filed this issue and I don't have that info available anymore.

Understood.  Thanks for getting back to me anyway.

**Comment by dpsutton** [comment]
The workaround for this is quite confusing. Hiding a table and then unhiding the table will trigger a table sync. But each of these constituent parts runs in the normal sync of a database. If normal sync "gets confused" i cannot imagine how an individual table sync would get confused. Is this issue still present? Anyone still seeing this?

**Comment by flamber** [comment]
Closing this issue, since it was a mix of BigQuery (originally), then Redshift and MongoDB stuff mixed in.

The issue should be completely fixed for BigQuery and Redshift in 0.41, while Mongo has it's own issues, which might have some of the same error message, but it very different by nature - list to known Mongo sync issues:
https://github.com/metabase/metabase/issues?q=is%3Aissue+is%3Aopen+label%3ADatabase%2FMongo+label%3AType%3ABug+label%3A%22Administration%2FMetadata+%26+Sync%22
